gio/src/task.rs

```diff
@@ -60,10 +86,156 @@ impl Task {
+            ))
+        }
+    }
+
+    #[doc(alias = "g_task_is_valid")]
+    pub fn is_valid(
+        result: &impl IsA<AsyncResult>,
+        source_object: Option<&impl IsA<glib::Object>>,
+    ) -> bool {
+        unsafe {
+            from_glib(ffi::g_task_is_valid(
+                result.as_ref().to_glib_none().0,
+                source_object.map(|p| p.as_ref()).to_glib_none().0,
+            ))
+        }
+    }
+
+    #[doc(alias = "completed")]
+    pub fn connect_completed_notify<F: Fn(&Self) + 'static>(&self, f: F) -> SignalHandlerId {
+        unsafe extern "C" fn notify_completed_trampoline<V, F>(
+            this: *mut ffi::GTask,
+            _param_spec: glib::ffi::gpointer,
+            f: glib::ffi::gpointer,
+        ) where
+            V: ValueType,
+            F: Fn(&Task<V>) + 'static,
+        {
+            let f: &F = &*(f as *const F);
+            f(&from_glib_borrow(this))
+        }
+        unsafe {
+            let f: Box_<F> = Box_::new(f);
+            connect_raw(
+                self.as_ptr() as *mut _,
+                b"notify::completed\0".as_ptr() as *const _,
+                Some(transmute::<_, unsafe extern "C" fn()>(
+                    notify_completed_trampoline::<V, F> as *const (),
+                )),
+                Box_::into_raw(f),
+            )
+        }
+    }
+
+    #[doc(alias = "g_task_return_value")]
+    #[doc(alias = "g_task_return_boolean")]
+    #[doc(alias = "g_task_return_int")]
+    #[doc(alias = "g_task_return_pointer")]
+    #[doc(alias = "g_task_return_error")]
+    pub fn return_result(&self, result: Result<&V, glib::Error>) {
+        unsafe extern "C" fn value_free(value: *mut c_void) {
+            let _: glib::Value = from_glib_full(value as *mut glib::gobject_ffi::GValue);
+        }
+
+        match result {
+            Ok(v) => unsafe {
+                ffi::g_task_return_pointer(
+                    self.to_glib_none().0,
+                    v.to_value().to_glib_full() as *mut _,
+                    Some(value_free),
+                )
+            },
+            Err(e) => unsafe {
+                ffi::g_task_return_error(self.to_glib_none().0, e.to_glib_full() as *mut _);
+            },
+        }
+    }
+
+    #[doc(alias = "g_task_propagate_value")]
+    #[doc(alias = "g_task_propagate_boolean")]
+    #[doc(alias = "g_task_propagate_int")]
+    #[doc(alias = "g_task_propagate_pointer")]
+    pub fn propagate(self) -> Result<V, glib::Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let value = ffi::g_task_propagate_pointer(self.to_glib_none().0, &mut error);
+            if !error.is_null() {
+                return Err(from_glib_full(error));
+            }
+            let value =
+                Option::<glib::Value>::from_glib_full(value as *mut glib::gobject_ffi::GValue)
+                    .unwrap_or_else(|| glib::Value::from_type(glib::types::Type::UNIT));
+            Ok(V::from_value(&value))
```

Comment:
Please double check this... I am not convinced it is correct in the `else` case of the line above